mesonbuild/compilers/c.py

```diff
@@ -160,15 +160,8 @@ def get_allow_undefined_link_args(self):
         '''
         Get args for allowing undefined symbols when linking to a shared library
         '''
-        if self.id == 'clang':
-            if self.clang_type == compilers.CLANG_OSX:
-                # Apple ld
-                return ['-Wl,-undefined,dynamic_lookup']
-            else:
-                # GNU ld and LLVM lld
-                return ['-Wl,--allow-shlib-undefined']
```

Comment:
The 'allow-shlib-undefined' case seems to be dropped.  Intentional?

Reply:
huh? https://github.com/mesonbuild/meson/pull/4183/files#diff-4f050ed89f9a7e15b09dd12b1e940a19R170